code/game/gamemodes/cult/cult_structures.dm

```diff
@@ -17,58 +17,566 @@
+		return
+
+	sacrifice = victim
+	sacrificer = user
+	//Sacrifice accepted, display message here
+	speak_to(user, "Your sacrifice has been deemed worthy, and accepted. End its life now, and liberate its soul, to seal our contract..")
+	sacrifice << span("danger", "You feel an invisible force grip your soul, as you're drawn inexorably towards the pylon. Every part of you screams to flee from here!")
+
+	if (istype(sacrifice.loc,/obj/item/weapon/holder))
+		var/obj/item/weapon/holder/H = sacrifice.loc
+		H.release_to_floor()
+	else
+		sacrifice.forceMove(get_turf(sacrifice))
+	pylonmode = 1
+	update_icon()
+	start_process()
+
+
+
+//Runs every proc while there's a pending sacrifice. This checks whether the sacrifice was killed or escaped
+/obj/structure/cult/pylon/proc/handle_sacrifice()
+	.=0 //Return value has 3 states.
+	//0 = waiting, the sacrifice is still alive
+	//1 = Finished, the sacrifice was killed, finalize the process and become a turret
+	//-1 = Escape: The sacrifice left the vicinity of the pylon, return to inert mode.
+
+	if (!sacrifice)
+		//If the sacrifice was deleted somehow, we cant know exactly what happened. We'll assume it escaped
+		.=-1
+
+	else if(get_dist(src, sacrifice) > 5)
+		//If the sacrifice gets more than 5 tiles away, it has escaped
+		.=-1
+	else if (sacrifice.stat == DEAD)
+		.=1
+
+	switch(.)
+		if (1)
+			finalize_sacrifice()
+		if (-1)
+			speak_to(sacrificer, "Fool! Your offering has escaped. Bring it back, or find us another..")
+			sacrifice = null
+			pylonmode = 0
+			update_icon()
+			if (sacrifice)
+				walk_to(sacrifice,0)
+		else
+			walk_towards(sacrifice,src, 10)
+
+/obj/structure/cult/pylon/proc/finalize_sacrifice()
+	sacrifice.visible_message(span("danger","\The [sacrifice]'s physical form unwinds as its soul is extracted from the remains, and drawn into the pylon!"))
+	if (istype(sacrifice.loc,/obj/item/weapon/holder))
+		var/obj/item/weapon/holder/H = sacrifice.loc
+		H.release_to_floor()
+	else
+		sacrifice.forceMove(get_turf(sacrifice)) //Make sure its on the floor before we gib it
+	pylonmode = 2
+	sacrifice.gib()
+
+	update_icon()
+
+
+//Called every process in turret mode, and also by chaining spawns
+/obj/structure/cult/pylon/proc/handle_firing()
+
+	if ((world.time < next_shot) || isbroken)
+		return
+		//Not ready to fire yet
+
+
+	var/list/stuffcache
+	//Stuffcache holds a list of things found by a dview call, so we only need to do it once per proc.
+
+
+	if (target && (target.stat != DEAD))
+		if (target.loc == last_target_loc)
+		//To minimise expensive DVIEW calls, if the target hasnt moved since the last shot we'll assume they're still in sight
+		//This could cause problems in some edge cases (like lowering firelocks or shutters without moving)
+		//But it has a major performance gain that makes it worth it
+			fire_at(target)
+			return
+		else
+			stuffcache = list()
+			DVIEW(stuffcache, 10, get_turf(src), INVISIBILITY_LIGHTING)

```

Comment:
Because cult are evil, an like darkness. Notably, constructs have darkvision.
As a magical evil antag, i want to make sure darkness doesn't hurt them, in order to allow it to be used as a tactical advantage. Hence, Dview allows them to target things in the dark